### PR TITLE
pipebomb

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

image

removes IEDs, these are its replacement

image

this recipe produces a finished pipebomb

you can craft pipes (of the atmos kind) with some metal
then you can make it into a pair of halved pipes with a welder
then you stuff in items for extra effect (optional, only works if it does something)
then you put in welding fuel OR gunpowder (no other options as of now, gunpowder is a better explosive)
cable finishes that part
after that, you can attach an unsecured assembly to finish the bomb

also this is the math for the explosion, power is decided by fuel
image

using it in-hand activates the assembly, using it off-hand allows you to configure the assembly

## Why It's Good For The Game

IEDs suck and stuff and this would be a better upgrade/replacement because with assemblies you could make a circuit sentient pipebomb or gift a pipebomb via cargo or something and gimmicks i guess
## Changelog
:cl:
add: Pipebombs
del: Improvised Firebombs
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
:-1: change it back to a soda can and this is 👍

**Comment by Admiral-Gigaton** [comment]
> 👎 change it back to a soda can and this is 👍

The soda can is funny, but this IS a pipe bomb...

**Comment by Pickle-Coding** [comment]
The radius should be proportional to the square root of the power, so the area of the effects grows proportionally with the power.

**Comment by mc-oofert** [comment]
> The radius should be proportional to the square root of the power, so the area of the effects grows proportionally with the power.

sir there is no radius variable

**Comment by Pickle-Coding** [comment]
> > The radius should be proportional to the square root of the power, so the area of the effects grows proportionally with the power.
> 
> 
> 
> sir there is no radius variable

The explosion values you put into the explosion proc are the radius of those given effects.

**Comment by mc-oofert** [comment]
> > > The radius should be proportional to the square root of the power, so the area of the effects grows proportionally with the power.
> > 
> > 
> > sir there is no radius variable
> 
> The explosion values you put into the explosion proc are the radius of those given effects.

and i also dont want to do it

**Comment by Pickle-Coding** [comment]
Why not?

**Comment by SomeoneYouProbablyKnow** [comment]
> The soda can is funny, but this IS a pipe bomb...

Teh craft should use a pipe instead of metal

**Comment by HoodMilk** [comment]
Please use a pipe in the recipe instead of metal, assistants should get blown into walls at 5000 mph when trying to wrench apart pipes for their bombs.

**Comment by mc-oofert** [comment]
> Please use a pipe in the recipe instead of metal, assistants should get blown into walls at 5000 mph when trying to wrench apart pipes for their bombs.

the recipe is implying you used the iron sheet to craft a pipe (recipe already existed) and then crafted it

**Comment by SomeoneYouProbablyKnow** [comment]
Implies yes but if you put it as an actual pipe you still retain the ability to use metal but
make it so most people (who don't know/care about crafting pipes from metal) will instead rip out pipes from the ground. Which is fun and cool and makes atmos techs even more annoyed at assistants

**Comment by Higgin** [comment]
seconding "make it actually require a pipe"

**Comment by moocowswag** [comment]
making it require a pipe to skip an easy intermediate step is just gonna (very mildly) arbitrarily gatekeep it from new or lesser skilled/knowledgeable/persistent players to a degree and will make balancing it harder.

Arbitrary gatekeeping is how we end up with things where 4 people do something that literally anyone can do, then it gets buffed "because nobody is using it/it seems weak" and then next thing you know half the server is using it to blow eachother up.

**Comment by SpessManCN** [comment]
why people alway replace but not adding stuff?

**Comment by carlarctg** [comment]
Definitely need to rename it to pipebomb now

I do think this should be a separate, larger version of the IED but at least I won't have to go around desperately scrounging for soda cans now

**Comment by mc-oofert** [comment]
image

**Comment by mc-oofert** [comment]
> making it require a pipe to skip an easy intermediate step is just gonna (very mildly) arbitrarily gatekeep it from new or lesser skilled/knowledgeable/persistent players to a degree and will make balancing it harder. (These is very impactful when the sample size is literally every round on every server)
> 
> Arbitrary gatekeeping is how we end up with things where 4 people do something that literally anyone can do, then it gets buffed "because nobody is using it/it seems weak" and then next thing you know half the server is using it to blow eachother up.

i have played atleast 500 sybil rounds and like 10 on terry and on like 1% of them i saw firebombs being used

**Comment by atlasle** [comment]
genuinely why can't both exist

**Comment by jughu** [comment]
firebombs were spammed when elances were a thing beside methbombs because that detonared it on hit which makes it not so random anymore the randomness makes them used less but they are still a ok weapon though chem grenades are better and probaly easier to craft over the couple years. i think pipebombs are a proper replacement(though i would love what goofball said and have a rigged soda can)

**Comment by breadcrombs** [comment]
I don't see a infrared in the image, can this work with a infrared?

**Comment by mc-oofert** [comment]
> I don't see a infrared in the image, can this work with a infrared?

no

**Comment by L0pz** [comment]
I love the Idea but I also think the IEDs should be kept as a shitty meme item. Also, it would be really cool if pipe bombs either shot shrapnel in random directions, or just emedded it into people who were caught in the explosion, since that's how they work irl ss far as I know and it would make for a fun mechanic

**Comment by Pickle-Coding** [comment]
> Why not?

Mc-Oofert please elaborate and answer this. Almost every other explosion source (except some ancient shitcode instances) uses square root scaling for the explosion radii. It's nonsensical and inconsistent to leave it the way it is.

File: code/game/objects/items/grenades/ghettobomb.dm [hunk 1]
```diff
@@ -15,67 +18,274 @@
...
+			return
+		coil.use(15)
+
+		var/cur_power = 0
+		for(var/datum/reagent/reagent as anything in reagents.reagent_list)
+			if(!(reagent.type in fuel_power))
+				continue
+			cur_power += fuel_power[reagent.type] * reagent.volume / reagents.maximum_volume
+
+		power *= cur_power
```
Comment (Pickle-Coding) [inline code/game/objects/items/grenades/ghettobomb.dm:257]: Why not just set power to cur_power and instead scale up the fuel list?

**Comment by mc-oofert** [comment]
> > Why not?
> 
> Mc-Oofert please elaborate and answer this. 

I don't have to

**Comment by RobertRoberts2020** [comment]
I like this but why did ieds need removal exactly? They were supposed to "suck" and be cheap

**Comment by RobertRoberts2020** [comment]
Doesnt work with infrarred emitters

**Comment by SomeoneYouProbablyKnow** [comment]
Never seen an infrared used as an actual tripwire. People just use them for instat explosion throwabled. So fuck it, really

**Comment by RobertRoberts2020** [comment]
I mean I can workaround this limitation by using signalers but that's kinda complicated, infrarred emitter is an assembly and your instructions state " any assembly" so uh, you know

**Comment by RobertRoberts2020** [comment]
In addition, I made one with plasma the other day and literally nothing happened when it went off,

**Comment by mc-oofert** [comment]
> Doesnt work with infrarred emitters

intentional, i had an aneurysm everytime someone used infrared singletanks

> In addition, I made one with plasma the other day and literally nothing happened when it went off,

pipebombs follow a list of reagent to power multiplier to decide their range, if you feel something is missing PR it

**Pull request merged**